### PR TITLE
New serverless pattern - S3-Lambda-Bedrock-AOSS

### PR DESCRIPTION
**Pull request by rajavaid77**
*Issue #, if available:*

*Description of changes:*

Added a serverless document embeddings processing & storage pattern using S3, EventBridge, Lambda, Bedrock and OpenSearch Serverless with SAM (Python). Lambda function is triggered by S3 events upon document uploads. Lambda function invokes Bedrock to create embeddings for uploaded documents and then stores the embeddings in a vector index in OpenSearch Serverless collection.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by parikhudit** [comment]
@rajavaid77 Thanks for the PR. Could you remove all the dependency modules from code and add a requirements.txt file along with instructions to download/install the required modules?

**Comment by rajavaid77** [comment]
@parikhudit Thanks for the feedback. update the code to remove dependencies, provided script to install and package dependencies as well as remove unwanted files.

**Comment by rajavaid77** [comment]
> @rajavaid77 Thanks for the PR. Could you remove all the dependency modules from code and add a requirements.txt file along with instructions to download/install the required modules? Also kindly remove any files that are extra/not needed.

> @parikhudit Thanks for the feedback. update the code to remove dependencies, provided script to install and package dependencies as well as remove unwanted files.

File: s3-lambda-bedrock-aoss/src/document_ingestion.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    )
+    # create an opensearch client and use the request-signer
+    return OpenSearch(
+        hosts=[{'host': host, 'port': 443}],
+        http_auth=auth,
+        use_ssl=True,
+        verify_certs=True,
+        connection_class=RequestsHttpConnection
+    )
+
```
Comment (parikhudit) [inline s3-lambda-bedrock-aoss/src/document_ingestion.py:72]: Extra line

File: s3-lambda-bedrock-aoss/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# S3 to Lambda to OpenSearch Serverless with Bedrock for Document Embeddings
+
+This patterns can be used to automatically create embeddings using Amazon Bedrock for documents uploaded to an Amazon S3 bucket and store the embeddings along with the document content in an Amazon OpenSearch Serverless index.
```
Comment (parikhudit) [inline s3-lambda-bedrock-aoss/README.md:3]: pattern

File: s3-lambda-bedrock-aoss/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+This patterns is useful for RAG-based applications that use Amazon S3 as source for the knowledge base for their LLM models. The pattern helps to seamlessly create and store the embeddings for new documents once they are upload to S3.
```
Comment (parikhudit) [inline s3-lambda-bedrock-aoss/README.md:5]: pattern

File: s3-lambda-bedrock-aoss/README.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+Learn more about this pattern at Serverless Land Patterns: << Add the live URL here >>
```
Comment (parikhudit) [inline s3-lambda-bedrock-aoss/README.md:7]: Please add URL.. you can take reference from any other pull requests

File: s3-lambda-bedrock-aoss/template.yaml [hunk 5]
```diff
@@ -0,0 +1,229 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: Automating document vectorization using OpenSearch and Bedrock
+Parameters:
+#  AmazonOpenSearchHost:
+#    Type: String
+#    ConstraintDescription: "Must be a valid Amazon OpenSearch Service host."
+#    Description: "The host name of the Amazon OpenSearch Service. Example: example.us-east-1.es.amazonaws.com"
```
Comment (parikhudit) [inline s3-lambda-bedrock-aoss/template.yaml:8]: Could you please remove this block if not needed

File: s3-lambda-bedrock-aoss/src/document_ingestion.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    body = json.dumps({"inputText": content})
+    response = bedrock_client.invoke_model(
+        body=body, modelId=modelId, accept=accept, contentType=contentType
+    )
+    response_body = json.loads(response.get("body").read())
+    embedding = response_body.get("embedding")
+    return embedding
+
+
+#write a function to remove a prefix of https:// from the given string
```
Comment (parikhudit) [inline s3-lambda-bedrock-aoss/src/document_ingestion.py:51]: The comments doesn't match the code that's already written. e.g. could the comment here be something like "function to remove a prefix 'https://' from the given string" ?
Please check the same with other comments/files in the PR. Thanks.

File: s3-lambda-bedrock-aoss/src/document_ingestion.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    # create an opensearch client and use the request-signer
+    return OpenSearch(
+        hosts=[{'host': host, 'port': 443}],
+        http_auth=auth,
+        use_ssl=True,
+        verify_certs=True,
+        connection_class=RequestsHttpConnection
+    )
+    
+
```
Comment (parikhudit) [inline s3-lambda-bedrock-aoss/src/document_ingestion.py:73]: Code debt: could you update the code to have consistent new lines across to same number of new line(s) where required. e.g. there are 2 new lines here whereas there a single new line on line 81.

File: s3-lambda-bedrock-aoss/README.md (hunk 2 above)
Comment (rajavaid77) [reply s3-lambda-bedrock-aoss/README.md]: fixed

File: s3-lambda-bedrock-aoss/README.md (hunk 3 above)
Comment (rajavaid77) [reply s3-lambda-bedrock-aoss/README.md]: fixed

File: s3-lambda-bedrock-aoss/README.md (hunk 4 above)
Comment (rajavaid77) [reply s3-lambda-bedrock-aoss/README.md]: fixed

File: s3-lambda-bedrock-aoss/src/document_ingestion.py (hunk 6 above)
Comment (rajavaid77) [reply s3-lambda-bedrock-aoss/src/document_ingestion.py]: fixed

File: s3-lambda-bedrock-aoss/src/document_ingestion.py (hunk 7 above)
Comment (rajavaid77) [reply s3-lambda-bedrock-aoss/src/document_ingestion.py]: fixed

**Pull request merged**